gcs/COOPERATIVE_LOCKING.md

```diff
@@ -0,0 +1,106 @@
+### Introducing GCS Connector Cooperative Locking
+
+We are excited to release Cooperative Locking support for GCS Connector. With
+this release you will be able to isolate and recover directory modification
+operations performed through hadoop fs command line interface on Cloud Storage.
+
+The key motivation behind this feature is a fact that directory mutations are
+not atomic on GCS, which means that competing directory mutations are
+susceptible to race conditions that can lead to data integrity issues (loss,
+duplication, etc).
+
+The key benefits to this feature are:
+
+*   Write-only isolation of move and delete directory operations performed on
+    GCS through `hadoop fs` command line interface.
+*   Logging of modified objects in directory modification operations.
+*   Recovery of failed directory operations using FSCK command line tool.
+
+### Enabling Cooperative Locking
+
+To enable this feature set `fs.gs.cooperative.locking.enable` Hadoop property to
+true in `core-site.xml`:
+
+```xml
+<configuration>
+  <property>
+    <name>fs.gs.cooperative.locking.enable</name>
+    <value>true</value>
+  </property>
+</configuration>
+```
+
+or specify it in directly `hadoop fs` command:
+
+```shell
+hadoop fs -Dfs.gs.cooperative.locking.enable=true ...
+```
+
+### Using Cooperative Locking
```

Comment:
What I use the feature in Spark/Hadoop, if nothing needs to be done specially, still mention it explicitly.

Reply:
In "Limitations" section it's mentioned that Cooperative Locking intended for use by human operators.